### PR TITLE
fix(vscode): Dependency timeout

### PR DESCRIPTION
**Pull request by YogiJogiFresh**
Changes:
- Dependency Timeout setting default increased from **120** to **300** seconds
- Dependency Timeout will ask users to if they would like to retry
  - If users select "No" -> Will show notification with dependency download link in message
- Fix dependency timeout blocking sequential dependency installation
  - async timeout no longer throws an Error that is caught in validateAndInstallBinaries, thus breaking out and stopping the others.
  -  Will instead display information message asking for retry. 
- Fix Update Azure Function Core Tools dependency
  - Stop and start Design time process
- Fix unintended update to .vscode/settings.json
  - .vscode/settings.json was unintentionally modified in LogicAppsUX
  - Added a check for isLogicAppProject

### CODE REVIEW

File: apps/vs-code-designer/src/app/utils/codeless/startDesignTimeApi.ts [hunk 1]
```diff
@@ -190,12 +190,13 @@ export function startDesignTimeProcess(
 }
 
 export function stopDesignTimeApi(): void {
+  ext.outputChannel.appendLog('Stopping Design Time Api');
   if (ext.designChildProcess === null || ext.designChildProcess === undefined) {
     return;
   }
-
+  cp.exec('echo Stop DesignTime Api');
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/utils/codeless/startDesignTimeApi.ts:197]: Does this work for linux and windows?

File: apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts [hunk 2]
```diff
@@ -240,32 +241,34 @@ export async function setDotNetCommand(context: IActionContext): Promise<void> {
     try {
       if (vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) {
         const workspaceFolder = await getWorkspaceFolder(context);
-        const projectPath = await tryGetFunctionProjectRoot(context, workspaceFolder);
+        const folderPath = isString(workspaceFolder) ? workspaceFolder : workspaceFolder.uri.fsPath;
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts:244]: Just as a reminder on what we discussed. If you think this change will work better, I am good with it!

**Review by ccastrotrejo — APPROVED** [review]

File: apps/vs-code-designer/src/app/utils/codeless/startDesignTimeApi.ts (hunk 1 above)
Comment (YogiJogiFresh) [reply apps/vs-code-designer/src/app/utils/codeless/startDesignTimeApi.ts:197]: ah, I need to remove this. This was for debugging purposes. Changed it to ext.outputChannel

File: apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts (hunk 2 above)
Comment (YogiJogiFresh) [reply apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts:244]: Reverting

File: apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts [hunk 3]
```diff
@@ -260,6 +261,7 @@ export async function setDotNetCommand(context: IActionContext): Promise<void> {
             }
 
             case Platform.mac: {
+              ext.logicAppWorkspace;
```
Comment (ccastrotrejo) [inline apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts:264]: Does this do something?

**Review by ccastrotrejo — APPROVED** [review]

File: apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts (hunk 3 above)
Comment (YogiJogiFresh) [reply apps/vs-code-designer/src/app/utils/dotnet/dotnet.ts:264]: oops. nope. will remove

**Pull request merged**